client/servicediscovery/tso_service_discovery.go

```diff
@@ -583,28 +565,29 @@ func (c *tsoServiceDiscovery) getTSOServer(sd ServiceDiscovery) (string, error)
 		err  error
 	)
 	t := c.tsoServerDiscovery
```

Comment:
Now this `t` is assigned across multiple lines in advance, which looks somewhat strange.

Reply:
assign it after calling discoverMicroservice now